docs/source/integrations/huggingface.rst

```diff
@@ -891,3 +901,1025 @@ model:
+
+The `description` argument can be used for whatever you like. When the dataset
+is loaded from the Hub, this description will be accessible via the dataset's
+:meth:`description <fiftyone.core.dataset.Dataset.description>` property.
+
+Additionally, you can specify the "format" of the uploaded dataset. By default,
+the format is the standard :ref:`FiftyOneDataset <FiftyOneDataset-import>` format,
+but you can also specify the data is uploaded in any of these
+:ref:`common formats <supported-import-formats>`. For example, to push the
+quickstart dataset in :ref:`COCO <COCODetectionDataset-import>` format, with a
+Creative Commons Attribution 4.0 license, you can do the following:
+
+.. code-block:: python
+    :linenos:
+
+    import fiftyone as fo
+    import fiftyone.zoo as foz
+    import fiftyone.utils.huggingface as fouh
+    import fiftyone.types as fot
+
+    dataset = foz.load_zoo_dataset("quickstart")
+    dataset_type = fot.dataset_types.COCODetectionDataset
+
+    fouh.push_to_hub(
+        dataset,
+        "quickstart-coco",
+        dataset_type=dataset_type,
+        license="cc-by-4.0",
+        label_fields="*" ### convert all label fields, not just ground truth
+    )
+
+
+.. note::
+
+    The `label_fields` argument is used to specify which label fields to convert
+    to the specified dataset type. By default when using some dataset formats,
+    only the `ground_truth` label field is converted. If you want to convert all
+    label fields, you can set `label_fields="*"`. If you want to convert specific
+    label fields, you can pass a list of field names.
+
+
+.. _huggingface-hub-load-dataset:
+
+Loading datasets from the Hub
+-----------------------------
+
+To load a dataset from the Hugging Face Hub, you can use the
+:func:`load_from_hub() <fiftyone.utils.huggingface.load_from_hub>` function.
+This function supports loading datasets in any of the 
+:ref:`common formats <supported-import-formats>` supported by FiftyOne, as well
+as image-based datasets stored via `Parquet <https://parquet.apache.org/>`_ files,
+as is common with datasets from the
+`datasets <https://huggingface.co/docs/datasets/en/index>`_ library which have
+been uploaded to the Hugging Face Hub. Below, we will walk through all of the
+ways you can load datasets from the Hub.
+
+In its simplest usage, the :func:`load_from_hub() <fiftyone.utils.huggingface.load_from_hub>`
+function only requires the `repo_id` of the dataset you want to load. For example,
+to load the :ref:`private dataset <huggingface-hub-push-dataset-advanced>` that
+we pushed to the Hub earlier, you can do the following:
+
+.. code-block:: python
+    :linenos:
+
+    import fiftyone.utils.huggingface as fouh
+
+    dataset = fouh.load_from_hub("<username-or-org>/my-private-dataset")
+
+
+.. note::
+
+    As long as you have an environment variable `HF_TOKEN` set with your Hugging
+    Face token (with read access), you can load gated datasets that you have
```

Comment:
```suggestion
    Face token (with read access), you can load private or gated datasets that you have
```

Reply:
We have private and gated datasets that need approval/agreement of some terms before accessing. The token will work the same either way but this might be clearer for end users.